core/src/main/java/kafka/server/share/DelayedShareFetch.java

```diff
@@ -204,6 +187,58 @@ Map<TopicIdPartition, FetchRequest.PartitionData> acquirablePartitions() {
         return topicPartitionData;
     }
 
+    /**
+     * Prepare partitions fetch data structure for acquirable partitions in the share fetch request satisfying minBytes criteria.
+     */
+    Map<TopicIdPartition, FetchPartitionData> replicaManagerFetchData(Map<TopicIdPartition, FetchRequest.PartitionData> topicPartitionData,
+                                                                      boolean hasRequestTimedOut) {
+        log.trace("Fetchable share partitions data: {} with groupId: {} fetch params: {}", topicPartitionData,
+            shareFetchData.groupId(), shareFetchData.fetchParams());
+        Map<TopicIdPartition, FetchPartitionData> replicaManagerFetchSatisfyingMinBytes = new HashMap<>();
```

Comment:
In most scenarios the request might have minBytes, hence do you always want to initialize a hash map? Mostly it will be overriden with `responseData` map. So can't it be null? Moreover can't it be simpy a boolean variable i.e. 

boolean minBytesSatisfied = false

if (accumulatedBytes.get() >= shareFetchData.fetchParams().minBytes)
                replicaManagerFetchSatisfyingMinBytes = responseData;
=>
if (accumulatedBytes.get() >= shareFetchData.fetchParams().minBytes)
                minBytesSatisfied = true;

if (replicaManagerFetchSatisfyingMinBytes.isEmpty() && !hasRequestTimedOut) { 
=>
if (!minBytesSatisfied && !hasRequestTimedOut) {

return replicaManagerFetchSatisfyingMinBytes;
=>
return Collections.emptyMap()

Reply:
yeah, it makes sense. I'll make the change.